### DISCUSSION TITLE
Failed to build physx

### DISCUSSION THREAD

**Discussion by luis605** [NVIDIA-Omniverse/PhysX#149] [General]
Does anyone knows how to correctly build Physx 5.1.3 in linux? I am getting this error and I have no idea how to fix it.
```cpp
CMake Error at /home/lixt/Desktop/projects/Lit Engine - engine/PhysX-104.2-physx-5.1.3/physx/snippets/compiler/cmake/linux/SnippetRender.cmake:32 (find_package):
  Could not find a package configuration file provided by "OpenGL" with any
  of the following names:

    OpenGLConfig.cmake
    opengl-config.cmake

  Add the installation prefix of "OpenGL" to CMAKE_PREFIX_PATH or set
  "OpenGL_DIR" to a directory containing one of the above files.  If "OpenGL"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  /home/lixt/Desktop/projects/Lit Engine - engine/PhysX-104.2-physx-5.1.3/physx/snippets/compiler/cmake/SnippetRender.cmake:32 (include)
  /home/lixt/Desktop/projects/Lit Engine - engine/PhysX-104.2-physx-5.1.3/physx/snippets/compiler/cmake/CMakeLists.txt:111 (INCLUDE)

```

I am generating my project with this command: `./generate_projects.sh linux-aarch64`. I am new to physx :sweat_smile: 
I also tried to `sudo  apt-get install libgl1-mesa-dev`

**Comment by preist-nvidia** [comment]
Hi @luis605 - did you check out the platform readme? https://github.com/NVIDIA-Omniverse/PhysX/blob/release/104.2/physx/documentation/platformreadme/linux/README_LINUX.md

**Reply by luis605** [reply]
I didn't. I will check it out soon. Thank you

**Comment by luis605** [comment]
I fixed the issue by running:

find / -name "opengl-config.cmake" 2>/dev/null
/home/lixt/packman-repo/chk/opengl-linux/2017.5.19.1/opengl-config.cmake


and then: export OpenGL_DIR=/home/lixt/packman-repo/chk/opengl-linux/2017.5.19.1/


Thank you @preist-nvidia for your help

**Reply by preist-nvidia** [reply]
Thank you. We'll double check this on our side, there should be either a note in the platform readme, or no env var setup needed.

**Comment by luis605** [comment]
I will close this discussion if I don't find any more error when compiling
